### DISCUSSION TITLE
Shader source code

### DISCUSSION THREAD

**Discussion by rolandhill** [vsg-dev/VulkanSceneGraph#606] [Q&A] [answered]
Hi Robert,

I need to check how something is calculated in the PBR shader and notice that now the shaders are loaded as compiled binary (pbr_ShaderSet.cpp). I can't find the source code anywhere in the VSG directories. Is the source code in there somewhere? I can't help feel that I'm just missing something obvious.

Regards,
Roland

**Comment by robertosfield** [comment, marked as answer]
Hi Roland,

The source shaders are where they used to be [vsgExamples/data/shaders](https://github.com/vsg-dev/vsgExamples/tree/master/data/shaders).

The new [vsgshadetset](https://github.com/vsg-dev/vsgExamples/tree/master/examples/utils/vsgshaderset) example loads the GLSL files for each ShadetSet and
then generates the appropriate SPIRV code and .cpp files to enable them to
be compiled in.

Cheers,
Robert.

**Comment by robertosfield** [comment]
In the old vsg-users google group there is a thread on vsgshaderset development:

[Improvements to built in ShaderSet SPIR-V variants for building without GLSLang and refinements to shader composition](https://groups.google.com/g/vsg-users/c/VEENdu2gdbc/m/QqFv3oGaCAAJ)

**Comment by rolandhill** [comment]
Thanks, Robert. I thought I must have been missing something obvious.
Thanks too for the link to the discussion.

Regards,
Roland
